label_studio/data_manager/actions/predictions_to_annotations.py

```diff
@@ -58,6 +58,13 @@ def predictions_to_annotations(project, queryset, **kwargs):
         )
         # Update counters for tasks and is_labeled. It should be a single operation as counters affect bulk is_labeled update
         project.update_tasks_counters_and_is_labeled(Task.objects.filter(id__in=tasks_ids))
+
+    try:
```

Comment:
it's better to move this block behind `if db_anntations`

Reply:
Also, it's better to check how TaskSerializerBulk.post_process_annotations() works. Probably, it makes sense to move recalculate_stats_async_or_sync inside of TaskSerializerBulk.post_process_annotations(). And it can solve other similar problems. Try searching where post_process_annotations() is used to understand it.